internal/txcommon/contract_inputs.go

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2023 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txcommon
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hyperledger/firefly-common/pkg/fftypes"
+	"github.com/hyperledger/firefly-common/pkg/i18n"
+	"github.com/hyperledger/firefly/pkg/core"
+)
+
+type BatchPinData struct {
+	Batch      *core.BatchPersisted `json:"batch"`
+	Contexts   []*fftypes.Bytes32   `json:"contexts"`
+	PayloadRef string               `json:"payloadRef"`
+}
+
+type BlockchainInvokeData struct {
+	Request  *core.ContractCallRequest `json:"request"`
+	BatchPin *BatchPinData             `json:"batchPin"`
+}
+
+func RetrieveBlockchainInvokeInputs(ctx context.Context, op *core.Operation) (*core.ContractCallRequest, error) {
+	var req core.ContractCallRequest
+	s := op.Input.String()
+	if err := json.Unmarshal([]byte(s), &req); err != nil {
+		return nil, i18n.WrapError(ctx, err, i18n.MsgJSONObjectParseFailed, s)
+	}
+	return &req, nil
+}
+
+func OpBlockchainInvoke(op *core.Operation, req *core.ContractCallRequest, batch *BatchPinData) *core.PreparedOperation {
```

Comment:
It's an anomaly to expose this builder to the world (normally they're "owned" by a single manager), but I didn't see a good way around it now that both Contract Manager and Multiparty Manager need the ability to trigger an invoke operation.